util.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+__license__ = """
+This file is part of OpenModelica.
+Copyright (c) 1998-CurrentYear, Open Source Modelica Consortium (OSMC),
+c/o Linköpings universitet, Department of Computer and Information Science,
+SE-58183 Linköping, Sweden.
+
+All rights reserved.
+
+THIS PROGRAM IS PROVIDED UNDER THE TERMS OF GPL VERSION 3 LICENSE OR
+THIS OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+ACCORDING TO RECIPIENTS CHOICE.
+
+The OpenModelica software and the Open Source Modelica
+Consortium (OSMC) Public License (OSMC-PL) are obtained
+from OSMC, either from the above address,
+from the URLs: http://www.ida.liu.se/projects/OpenModelica or
+http://www.openmodelica.org, and in the OpenModelica distribution.
+GNU version 3 is obtained from: http://www.gnu.org/copyleft/gpl.html.
+
+This program is distributed WITHOUT ANY WARRANTY; without
+even the implied warranty of  MERCHANTABILITY or FITNESS
+FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
+
+See the full OSMC Public License conditions for more details.
+"""
+
+def pythonBoolToModelicaBool(value):
+  if value:
+    return "true"
+  else:
+    return "false"
+
+def wordsBeforeAfterLastDot(value, lastWord):
+  if not value:
+    return ""
+
+  value = value.strip()
+  pos = 0
+  if value.endswith('\''):
+    i = len(value) - 2
+    while value[i] != '\'' and i > 1 and value[i-1] != '\\':
+      i -= 1
+
+    pos = i - 1
+  else:
+    pos = value.rfind('.')
+
+  if pos >= 0:
+    if lastWord:
+      return value[pos+1:len(value)]
+    else:
+      return value[0:pos]
+  else:
+    return value
+
+def getLastWordAfterDot(value):
+  return wordsBeforeAfterLastDot(value, True)
+
+def removeFirstLastCurlBrackets(value):
+  value = value.strip()
+  if (len(value) > 1 and value[0] == '{' and value[len(value) - 1] == '}'):
+    value = value[1: len(value) - 1]
+  return value
+
+def removeFirstLastParentheses(value):
+  value = value.strip()
+  if (len(value) > 1 and value[0] == '(' and value[len(value) - 1] == ')'):
+    value = value[1: len(value) - 1]
+  return value
+
+def unparseArrays(value):

```

Comment:
Please add tests for all functions in this file. The tests will aid the reader in understanding what these functions do - by showing what output is expected for a given input. This will be quite handy because at the moment the functions are quite involved and hence not easy to understand/read what they do without spending a non-trivial amount of time reading the function body.